ios/ReactNativePageView.m

```diff
@@ -1,11 +1,216 @@
 
 #import "ReactNativePageView.h"
 #import "React/RCTLog.h"
+#import <React/RCTViewManager.h>
+
+
+@interface RCTOnPageScrollEvent : NSObject <RCTEvent>
+
+- (instancetype) initWithReactTag:(NSNumber *)reactTag
+                         position:(NSNumber *)position
+                           offset:(NSNumber *)offset
+                    coalescingKey:(uint16_t)coalescingKey;
+
+@end
+
+@implementation RCTOnPageScrollEvent
+{
+    NSNumber* _position;
+    NSNumber* _offset;
+    uint16_t _coalescingKey;
+}
+
+@synthesize viewTag = _viewTag;
+
+- (NSString *)eventName {
+    return @"onPageScroll";
+}
+
+- (instancetype) initWithReactTag:(NSNumber *)reactTag
+                         position:(NSNumber *)position
+                           offset:(NSNumber *)offset
+                    coalescingKey:(uint16_t)coalescingKey;
+{
+    RCTAssertParam(reactTag);
+    
+    if ((self = [super init])) {
+        _viewTag = reactTag;
+        _position = position;
+        _offset = offset;
+        _coalescingKey = coalescingKey;
+    }
+    return self;
+}
+
+RCT_NOT_IMPLEMENTED(- (instancetype)init)
+- (uint16_t)coalescingKey
+{
+    return _coalescingKey;
+}
+
+
+- (BOOL)canCoalesce
+{
+    return NO;
+}
+
++ (NSString *)moduleDotMethod
+{
+    return @"RCTEventEmitter.receiveEvent";
+}
+
+- (NSArray *)arguments
+{
+    return @[self.viewTag, RCTNormalizeInputEventName(self.eventName), @{
+                 @"position": _position,
+                 @"offset": _offset
+                 }];
+}
+
+- (id<RCTEvent>)coalesceWithEvent:(id<RCTEvent>)newEvent;
```

Comment:
This is optional and needs to be implemented only if `canCoalesce` returns `YES`